### DISCUSSION TITLE
plotting grains diameter histogram from different samples on one graph without having overlays of bins

### DISCUSSION THREAD

**Discussion by nargesmbg** [mtex-toolbox/mtex#1899] [Ask Anything]
hi, 

I have calculated grains for different samples and now I want to compare normalized histogram of them in one graph without having overlays of bins.  

I know I should use bar() in this case but the problem is the grain diameter size range of histogram is different for each samples. 
could you please help me with that? 
I also tried the following code to have a cropped diameter size range between 0 and 1000 um. this also doesn't work and it just plots one histogram (I guess it is h1) and it's not even groupping the data 
h1 = histogram(grains1, 'Normalization', 'probability');
h2 = histogram(grains2,'Normalization', 'probability');
h3 = histogram(grains3.diameter,'Normalization', 'probability');
h4 = histogram(grains4.diameter,'Normalization', 'probability');

x = 0:50:1000
bar(x,[h1.Values; h2.Values]', 'grouped')

thanks for your support in advance

**Comment by kilir** [comment]
Hi,
here is an example:
```
s1 = 100 + 50 *randn(1000,1);
s2 = 150 + 10 *randn(1000,1);
bw=10;
edges=[0:bw:200];
bc = edges(2:end)-bw/2;
n1 = histcounts(s1,edges, 'Normalization', 'probability');
n2 = histcounts(s2,edges, 'Normalization', 'probability');

bar(bc,[n1; n2], 'grouped')
```
Cheers,
Rüdiger
